torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2559,10 +2594,6 @@ def track_symint(source, val, constraint=None):
                             # expect to have to compile in this case anyway
                             if i not in (0, 1):
                                 constraint_violated = True
-                        else:
-                            # TODO: Maybe non-strict constraint shouldn't error
-                            # here?  Check what happens in practice
-                            constraint_violated = True
```

Comment:
hmm why do we no longer need this condition?

Reply:
Good question, and it's a difficult answer. This branch is triggered when:
* mark_dynamic() is manually called on a dim. This includes where it's done in the NT constructor and in the test with `DoubleSizeMaybeAddGeThreeTensor`. The manual `mark_dynamic()` results in a `RelaxedUnspecConstraint(warn_only=False)` for this dim.
* there is some symbolic relationship between the inner and outer tensor shapes. In the `DoubleSizeMaybeAddGeThreeTensor` case, outer = 2 * inner.

It feels wrong to say a constraint is violated whenever some symbolic relationship exists, so I removed this branch. Sure, it's not truly "unrestricted dynamic" because of the guard. This part is probably one of the sketchier parts of this PR, so I pinged @voznesenskym about this too to see if he's okay with this change.

Reply:
update: seems good